--- galeon-2.0.1/configure.in~	2006-02-27 11:16:34.000000000 +0100
+++ galeon-2.0.1/configure.in	2006-02-27 11:17:17.000000000 +0100
@@ -230,7 +230,7 @@
 MOZILLA_INCLUDE_ROOT="`$PKG_CONFIG --variable=includedir $MOZILLA-xpcom`"
 AC_SUBST(MOZILLA_INCLUDE_ROOT)
 
-MOZILLA_HOME="`$PKG_CONFIG --variable=libdir $MOZILLA-xpcom`"
+MOZILLA_HOME="`$PKG_CONFIG --variable=libdir $MOZILLA-plugin`"
 AC_SUBST(MOZILLA_HOME)
 
 PKG_CHECK_MODULES(MOZILLA_COMPONENT, $MOZILLA-gtkmozembed >= $min_version,
