diff -urN galeon-0.11.0.org/src/main.c galeon-0.11.0/src/main.c
--- galeon-0.11.0.org/src/main.c	Sat Jun 30 23:36:45 2001
+++ galeon-0.11.0/src/main.c	Sat Jun 30 23:38:06 2001
@@ -41,6 +41,7 @@
 #include <applet-widget.h>
 #endif
 
+#include <stdlib.h>
 #include <errno.h>
 
 /* local function prototypes */
@@ -132,6 +133,10 @@
 	gint i, n_urls;
 	gchar **url;
 
+        /* Setting MOZILLA_FIVE_HOME */
+	if (getenv("MOZILLA_FIVE_HOME") == NULL)
+	    setenv("MOZILLA_FIVE_HOME", "/usr/X11R6/lib/mozilla/", 1);
+	
 	 /* make stdout line buffered - we only use it for debug info */
 	setvbuf (stdout, NULL, _IOLBF, 0);
 
