diff -aurN galeon-1.3.10.orig/galeon.desktop.in galeon-1.3.10/galeon.desktop.in
--- galeon-1.3.10.orig/galeon.desktop.in	2003-09-19 07:00:22.000000000 +0200
+++ galeon-1.3.10/galeon.desktop.in	2003-11-07 00:49:49.000000000 +0100
@@ -8,7 +8,7 @@
 Terminal=false
 Type=Application
 Icon=galeon
-Categories=Application;Network;
+Categories=GTK;GNOME;Network;WebBrowser;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=galeon
 X-GNOME-Bugzilla-Component=general
