diff -urN galeon-1.3.5.orig/galeon.desktop.in galeon-1.3.5/galeon.desktop.in
--- galeon-1.3.5.orig/galeon.desktop.in	2003-02-21 23:41:35.000000000 +0100
+++ galeon-1.3.5/galeon.desktop.in	2003-06-26 08:04:41.000000000 +0200
@@ -7,7 +7,7 @@
 Terminal=0
 Type=Application
 Icon=galeon.png
-Categories=Application;Network;
+Categories=GTK;GNOME;Application;Network;X-WWW
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=galeon
 X-GNOME-Bugzilla-Component=general
