--- galeon-0.10.1/Makefile.am	Sun Feb 18 23:08:02 2001
+++ galeon-0.10.1/Makefile.am	Mon Feb 26 19:42:27 2001
@@ -9,7 +9,7 @@
 oaf_DATA = galeon.oafinfo
 oafdir = $(datadir)/oaf
 
-Networkdir = $(datadir)/gnome/apps/Internet
+Networkdir = $(datadir)/applnk/Network/WWW
 Network_DATA = $(DESKTOP_FILES)
 
 MyPortalCSS_DATA = myportal.css
--- galeon-0.10.1/galeon.desktop	Sat Feb 24 06:25:28 2001
+++ galeon-0.10.1/galeon.desktop	Mon Feb 26 19:43:09 2001
@@ -17,6 +17,7 @@
 Name[lt]=Galeon
 Name[nl]=Galeon
 Name[no]=Galeon
+Name[pl]=Galeon
 Name[pt_BR]=Galeon
 Name[ro]=Galeon
 Name[ru]=Галеон
