diff -urN aa/galeon-1.3.17/mozilla/GaleonWrapper.cpp galeon-1.3.17/mozilla/GaleonWrapper.cpp
--- aa/galeon-1.3.17/mozilla/GaleonWrapper.cpp	2004-08-07 19:09:26.000000000 +0200
+++ galeon-1.3.17/mozilla/GaleonWrapper.cpp	2004-09-16 20:41:42.521789744 +0200
@@ -571,7 +571,7 @@
 }
 
 #if !(MOZILLA_CHECK_VERSION4 (1,8,MOZILLA_ALPHA,3) || \
-      (MOZILLA_IS_BRANCH (1,7) && MOZILLA_CHECK_VERSION3(1,7,3)))
+      (MOZILLA_IS_BRANCH (1,7) && MOZILLA_CHECK_VERSION3(1,7,4)))
 /* Workaround for broken reload with frames, see mozilla bug
  * http://bugzilla.mozilla.org/show_bug.cgi?id=246392
  */
diff -urN aa/galeon-1.3.17/mozilla/GaleonWrapper.h galeon-1.3.17/mozilla/GaleonWrapper.h
--- aa/galeon-1.3.17/mozilla/GaleonWrapper.h	2004-08-07 19:09:26.000000000 +0200
+++ galeon-1.3.17/mozilla/GaleonWrapper.h	2004-09-16 20:42:08.623821632 +0200
@@ -95,7 +95,7 @@
 	nsresult GetDocumentUrl (nsCString &url);
 
 #if !(MOZILLA_CHECK_VERSION4 (1,8,MOZILLA_ALPHA,3) || \
-      (MOZILLA_IS_BRANCH (1,7) && MOZILLA_CHECK_VERSION3(1,7,3)))
+      (MOZILLA_IS_BRANCH (1,7) && MOZILLA_CHECK_VERSION3(1,7,4)))
 	nsresult ReloadPage (GtkMozEmbedReloadFlags reloadFlags);
 #endif
 	nsresult ReloadFrame ();
diff -urN aa/galeon-1.3.17/mozilla/mozilla-embed.cpp galeon-1.3.17/mozilla/mozilla-embed.cpp
--- aa/galeon-1.3.17/mozilla/mozilla-embed.cpp	2004-08-07 19:09:26.000000000 +0200
+++ galeon-1.3.17/mozilla/mozilla-embed.cpp	2004-09-16 20:45:10.782129360 +0200
@@ -961,7 +961,7 @@
 
 	
 #if MOZILLA_CHECK_VERSION4 (1,8,MOZILLA_ALPHA,3) || \
-    (MOZILLA_IS_BRANCH (1,7) && MOZILLA_CHECK_VERSION3(1,7,3))
+    (MOZILLA_IS_BRANCH (1,7) && MOZILLA_CHECK_VERSION3(1,7,4))
 	gtk_moz_embed_reload (GTK_MOZ_EMBED(embed), reloadFlags);
 	return G_OK;
 #else
