diff -urN galeon-0.4.org/Makefile.am galeon-0.4/Makefile.am
--- galeon-0.4.org/Makefile.am	Sat Sep  9 23:18:27 2000
+++ galeon-0.4/Makefile.am	Sat Sep  9 23:19:03 2000
@@ -14,11 +14,11 @@
 	    fi \
 	  done \
 	fi
-	$(INSTALL_PROGRAM) $(srcdir)/src/start_galeon $(bindir)
-	$(mkinstalldirs) $(datadir)/galeon
-	$(INSTALL_DATA) $(srcdir)/src/bookmarks.xml $(datadir)/galeon/bookmarks.xml
-	$(mkinstalldirs) $(datadir)/gnome/apps/Internet
-	$(INSTALL_DATA) $(srcdir)/src/galeon.desktop $(datadir)/gnome/apps/Internet	
+	$(INSTALL_PROGRAM) $(srcdir)/src/start_galeon $(DESTDIR)$(bindir)
+	$(mkinstalldirs) $(DESTDIR)$(datadir)/galeon
+	$(INSTALL_DATA) $(srcdir)/src/bookmarks.xml $(DESTDIR)$(datadir)/galeon/bookmarks.xml
+	$(mkinstalldirs) $(DESTDIR)$(datadir)/gnome/apps/Internet
+	$(INSTALL_DATA) $(srcdir)/src/galeon.desktop $(DESTDIR)$(datadir)/gnome/apps/Internet	
 dist-hook:
 	if test -d pixmaps; then \
 	  mkdir $(distdir)/pixmaps; \
