--- galeon-1.2.1/src/mozilla/ContentHandler.h.wiget	Tue Apr 23 10:39:08 2002
+++ galeon-1.2.1/src/mozilla/ContentHandler.h	Tue Apr 23 10:40:21 2002
@@ -31,6 +31,7 @@
 #include "nsCOMPtr.h"
 #include "nsISupports.h"
 #include "nsError.h"
+#include <libgnomevfs/gnome-vfs-mime-handlers.h>
 
 #define G_CONTENTHANDLER_CID			     \
 { /* 16072c4a-23a6-4996-9beb-9335c06bbeae */         \
@@ -41,7 +42,7 @@
 }
 
 class nsIFactory;
-struct GnomeVFSMimeApplication;
+//struct GnomeVFSMimeApplication;
 
 class GContentHandler : public nsIHelperAppLauncherDialog
 {
