diff -Nur galeon-0.11.1/configure.in galeon-0.11.1.new/configure.in
--- galeon-0.11.1/configure.in	Sun Jul  1 16:46:29 2001
+++ galeon-0.11.1.new/configure.in	Sun Jul  1 16:57:30 2001
@@ -320,7 +320,7 @@
 
 dnl Add the languages which your application supports here.
 ALL_LINGUAS="az ca cs da de es et fi fr gl hr hu it ja ko lt nl no pl pt_BR ro ru sk sl sv tr uk zh_TW.Big5 ms zh_CN.GB2312"
-AM_GNOME_GETTEXT
+AM_GNU_GETTEXT
 
 dnl Set PACKAGE_LOCALE_DIR in config.h.
 if test "x${prefix}" = "xNONE"; then
