diff -Nur galeon-0.11.0/src/galeon-wrapper.cpp galeon-0.11.0.new/src/galeon-wrapper.cpp
--- galeon-0.11.0/src/galeon-wrapper.cpp	Thu Jun  7 19:12:27 2001
+++ galeon-0.11.0.new/src/galeon-wrapper.cpp	Sun Jul  1 16:04:41 2001
@@ -443,11 +443,11 @@
 	nsCOMPtr<nsIDocument> doc = do_QueryInterface(aDOMDocument);
 	if(!doc) return NS_ERROR_FAILURE;
 
-	nsCOMPtr<nsIPresShell> presShell = getter_AddRefs(doc->GetShellAt(0));
-	if(!presShell) return NS_ERROR_FAILURE;
+//	nsCOMPtr<nsIPresShell> presShell = getter_AddRefs(doc->GetShellAt(0));
+//	if(!presShell) return NS_ERROR_FAILURE;
 
 	nsCOMPtr<nsIPresContext> presContext;
-	presShell->GetPresContext(getter_AddRefs(presContext));
+//	presShell->GetPresContext(getter_AddRefs(presContext));
 	if(!presContext) return NS_ERROR_FAILURE;
 
 	nsCOMPtr<nsISupports> pcContainer;
@@ -480,8 +480,8 @@
 	nsCOMPtr<nsIDocument> doc = do_QueryInterface(aDOMDocument);
 	if(!doc) return NS_ERROR_FAILURE;
 
-	nsIURI *uri = doc->GetDocumentURL();
-	uri->GetSpec (url);
+//	nsIURI *uri = doc->GetDocumentURL();
+//	uri->GetSpec (url);
 
 	return NS_OK;
 }
@@ -498,8 +498,8 @@
 	nsCOMPtr<nsIDocument> doc = do_QueryInterface(aDOMDocument);
 	if(!doc) return NS_ERROR_FAILURE;
 
-	nsIURI *uri = doc->GetDocumentURL();
-	uri->GetSpec (url);
+//	nsIURI *uri = doc->GetDocumentURL();
+//	uri->GetSpec (url);
 
 	return NS_OK;
 }
@@ -868,9 +868,9 @@
 	nsCOMPtr<nsIDocument> doc = do_QueryInterface (DOMDocument);
 	if (!doc) return NS_ERROR_FAILURE;
 	
-	nsIURI *uri = doc->GetDocumentURL ();
+//	nsIURI *uri = doc->GetDocumentURL ();
 	char *url;
-	uri->GetSpec (&url);
+//	uri->GetSpec (&url);
 	
 	nsCOMPtr<nsIPermissionManager> 
 		permissionManager = do_CreateInstance (PERMISSIONMANAGER_ID);
@@ -1075,9 +1075,9 @@
 				do_QueryInterface (aDOMDocument);
 			if(!doc) return NS_ERROR_FAILURE;
 			
-			nsIURI *uri = doc->GetDocumentURL();
-			result = uri->Resolve (link, favicon_url);
-			if (NS_FAILED (result)) return NS_ERROR_FAILURE;
+			//nsIURI *uri = doc->GetDocumentURL();
+			//result = uri->Resolve (link, favicon_url);
+			//if (NS_FAILED (result)) return NS_ERROR_FAILURE;
 			
 			g_free (link);
 			return NS_OK;
