diff -ur galeon-2.0.5.orig/mozilla/MozillaPrivate.cpp galeon-2.0.5/mozilla/MozillaPrivate.cpp
--- galeon-2.0.5.orig/mozilla/MozillaPrivate.cpp	2008-02-03 01:05:22.000000000 +0100
+++ galeon-2.0.5/mozilla/MozillaPrivate.cpp	2008-03-27 23:15:59.000000000 +0100
@@ -24,13 +24,11 @@
 #include "config.h"
 #endif
 
-#ifdef HAVE_NSSTRING_INTERNAL
-/***** BEGIN UNSAFE HEADERS ******/
 #define MOZILLA_INTERNAL_API
+/***** BEGIN UNSAFE HEADERS ******/
 #include <nsICSSLoader.h>
 #include <nsICSSStyleSheet.h>
 #include <nsIHTMLDocument.h>
-#endif
 
 #include <nsIPrintOptions.h>
 #include <nsIDocument.h>
@@ -40,10 +38,8 @@
 #include <nsStringAPI.h>
 
 #include "MozillaPrivate.h"
-#ifdef HAVE_NSSTRING_INTERNAL
-#undef MOZILLA_INTERNAL_API
 /***** END UNSAFE HEADERS ******/
-#endif
+#undef MOZILLA_INTERNAL_API
 
 #include <nsServiceManagerUtils.h>
 #include <nsISupportsPrimitives.h>
