diff -urN galeon-1.2.1.org/src/Makefile.am galeon-1.2.1/src/Makefile.am
--- galeon-1.2.1.org/src/Makefile.am	Sat Apr 20 20:17:58 2002
+++ galeon-1.2.1/src/Makefile.am	Sat Apr 20 20:18:20 2002
@@ -160,7 +160,7 @@
 	$(NAUTILUS_VIEW_SRC_TO_COMPILE)
 
 galeon_bin_LDADD = \
-	$(top_builddir)/src/mozilla/libmozilla-interaction.a \
+	$(top_builddir)/src/mozilla/libmozilla-interaction.la \
 	$(MOZILLA_COMPONENT_LDFLAGS) \
 	$(MOZILLA_COMPONENT_LIBS) \
 	$(GALEON_DEPENDENCY_LIBS) \
diff -urN galeon-1.2.1.org/src/mozilla/Makefile.am galeon-1.2.1/src/mozilla/Makefile.am
--- galeon-1.2.1.org/src/mozilla/Makefile.am	Sat Apr 20 20:17:58 2002
+++ galeon-1.2.1/src/mozilla/Makefile.am	Sat Apr 20 20:18:43 2002
@@ -100,9 +100,9 @@
 	gProtocolModule.cpp
 
 
-noinst_LIBRARIES = libmozilla-interaction.a
+noinst_LTLIBRARIES = libmozilla-interaction.la
 
-libmozilla_interaction_a_SOURCES = \
+libmozilla_interaction_la_SOURCES = \
 	mozilla.cpp			\
 	mozilla.h			\
 	BaseProtocolHandler.cpp		\
