--- galeon-1.3.5/galeon.schemas.in.orig	2003-07-04 18:17:32.000000000 +0200
+++ galeon-1.3.5/galeon.schemas.in	2003-07-04 18:36:25.000000000 +0200
@@ -643,7 +643,6 @@
         <applyto>/apps/galeon/UI/Toolbar/toolbar_style</applyto>
         <owner>galeon</owner>
         <type>int</type>
-        <default></default>
         <locale name="C">
         <short>Override toolbar style</short>
 	<long>If set, override the display of toolbar items from the desktop
