--- galeon-1.2.0/src/mozilla/Makefile.am.wiget	Mon Apr 15 18:02:59 2002
+++ galeon-1.2.0/src/mozilla/Makefile.am	Mon Apr 15 18:04:07 2002
@@ -94,9 +94,9 @@
 	gProtocolModule.cpp
 
 
-noinst_LIBRARIES = libmozilla-interaction.a
+noinst_LTLIBRARIES = libmozilla-interaction.la
 
-libmozilla_interaction_a_SOURCES = \
+libmozilla_interaction_la_SOURCES = \
 	mozilla.cpp			\
 	mozilla.h			\
 	BaseProtocolHandler.cpp		\
--- galeon-1.2.0/src/Makefile.am.wiget	Mon Apr 15 18:28:13 2002
+++ galeon-1.2.0/src/Makefile.am	Mon Apr 15 18:28:38 2002
@@ -160,7 +160,7 @@
 	$(NAUTILUS_VIEW_SRC_TO_COMPILE)
 
 galeon_bin_LDADD = \
-	$(top_builddir)/src/mozilla/libmozilla-interaction.a \
+	$(top_builddir)/src/mozilla/libmozilla-interaction.la \
 	$(MOZILLA_COMPONENT_LDFLAGS) \
 	$(MOZILLA_COMPONENT_LIBS) \
 	$(GALEON_DEPENDENCY_LIBS) \
