--- galeon-1.2.7/src/mozilla/GaleonWrapper.cpp.orig	Sat Oct 26 00:52:28 2002
+++ galeon-1.2.7/src/mozilla/GaleonWrapper.cpp	Fri Feb  7 13:51:12 2003
@@ -80,7 +80,9 @@
 #include "caps/nsIPrincipal.h"
 #include "nsIDeviceContext.h"
 #include "nsIPresContext.h"
+#if MOZILLA_SNAPSHOT > 1
 #include "nsITypeAheadFind.h"
+#endif
 #include "nsSupportsPrimitives.h"
 
 GaleonWrapper::GaleonWrapper ()
@@ -584,6 +586,7 @@
 			      PRBool search_in_frames,
 			      PRBool *didFind)
 {
+#if MOZILLA_SNAPSHOT > 1
 	if (!interactive)
 	{
 		nsresult rv;
@@ -610,6 +613,7 @@
 			}
 		}
 	}
+#endif
 
 	nsCOMPtr<nsIWebBrowserFind> finder (do_GetInterface(mWebBrowser));
 
