diff -Nur galeon-1.2.99.orig/configure.in galeon-1.2.99/configure.in
--- galeon-1.2.99.orig/configure.in	Mon Sep 16 00:04:20 2002
+++ galeon-1.2.99/configure.in	Wed Oct 16 22:46:52 2002
@@ -119,7 +119,7 @@
 MOZILLA_INCLUDE_ROOT="`$PKG_CONFIG --variable=includedir mozilla-gtkmozembed`"
 AC_SUBST(MOZILLA_INCLUDE_ROOT)
 
-MOZILLA_HOME="`$PKG_CONFIG --variable=libdir mozilla-gtkmozembed`"
+MOZILLA_HOME="`$PKG_CONFIG --variable=libdir mozilla-gtkmozembed`/mozilla"
 AC_SUBST(MOZILLA_HOME)
 
 dnl ################
diff -Nur galeon-1.2.99.orig/src/galeon.in galeon-1.2.99/src/galeon.in
--- galeon-1.2.99.orig/src/galeon.in	Fri Dec 28 01:13:42 2001
+++ galeon-1.2.99/src/galeon.in	Wed Oct 16 22:13:47 2002
@@ -2,31 +2,12 @@
 
 dist_bin=`dirname $0`
 
-if test -n "$MOZILLA_FIVE_HOME"; then
-	MOZILLA_HOME=$MOZILLA_FIVE_HOME
-elif [ -f @MOZILLA_HOME@/chrome/comm.jar ]; then
-	MOZILLA_HOME=@MOZILLA_HOME@
-elif [ -f /usr/local/mozilla/chrome/comm.jar ]; then
-	MOZILLA_HOME=/usr/local/mozilla
-elif [ -f /usr/lib/mozilla/chrome/comm.jar ]; then
-	MOZILLA_HOME=/usr/lib/mozilla
-elif [ -f /opt/mozilla/chrome/comm.jar ]; then
-	MOZILLA_HOME=/opt/mozilla
-elif [ -f /usr/lib/mozilla-0.9.7/chrome/comm.jar ]; then
-	MOZILLA_HOME=/usr/lib/mozilla-0.9.7/
-elif [ -f /opt/gnome/lib/mozilla/chrome/comm.jar ]; then
-        MOZILLA_HOME=/opt/gnome/lib/mozilla/
-else
-	echo "Cannot find mozilla installation directory. Please set MOZILLA_FIVE_HOME to your mozilla directory"
-	exit
-fi
+MOZILLA_FIVE_HOME=@MOZILLA_HOME@
+export MOZILLA_FIVE_HOME
 
-LD_LIBRARY_PATH=$MOZILLA_HOME:$LD_LIBRARY_PATH
+LD_LIBRARY_PATH=$MOZILLA_FIVE_HOME:$LD_LIBRARY_PATH
 export LD_LIBRARY_PATH
 
-MOZILLA_FIVE_HOME=$MOZILLA_HOME
-export MOZILLA_FIVE_HOME
-
 # for some reason this lets Galeon start properly on Red Hat 7.1 systems
 if [ -f /etc/redhat-release ]; then
 	if [ "`cat /etc/redhat-release`" = "Red Hat Linux release 7.1 (Seawolf)"  ]; then
